test/test_proxy.py

```diff
@@ -77,14 +77,28 @@ def test_ssl_context_with_proxy_is_respected(self):
         self.assertIn('SSL', str(cm.exception))
         self.assertEqual(1, len(self.proxy_server.requests))
 
-    def test_geocoder_constructor_uses_str_proxy(self):
+    def test_geocoder_constructor_uses_str_http_proxy(self):
         base_http = urlopen(self.remote_website_http, timeout=self.timeout)
         base_html = base_http.read()
-        geocoder_dummy = DummyGeocoder(proxies=self.proxy_url,
+        geocoder_dummy = DummyGeocoder(proxies=self.proxy_url, scheme='http',
```

Comment:
No, I don't think we need to test two schemes in two almost identical integration tests, which aren't cheap with regards to time. 

It would be better to keep the previous single test, and add a new *unit* test here. Something like this:
```
g = DummyGeocoder(proxies=self.proxy_url, scheme='http')
self.assertDictEqual(g.proxies, {'http': self.proxy_url, 'https': self.proxy_url})
```

That should be enough. We already tested that the `g.proxies` dict is respected in the integration tests, but we didn't check if both schemes are respected – which is exactly what this test does.

Reply:
I am not sure if I get this right, but this will assert error:

`self.assertDictEqual(g.proxies, {'http': self.proxy_url, 'https': self.proxy_url})`

So I added 2 DummyGeocoders to check if both schemes are respected.

Reply:
Have you by chance missed this comment? https://github.com/geopy/geopy/pull/300#discussion_r193676957

Reply:
Also it would've been better if it was in a separate `test_*` function.

Reply:
Oh yeah, I did miss this comment, pardon my inattention.